lib/lograge.rb

```diff
@@ -121,11 +121,18 @@ def unsubscribe(component, subscriber)
     events = subscriber.public_methods(false).reject { |method| method.to_s == 'call' }
     events.each do |event|
       Lograge.notification_listeners_for("#{event}.#{component}").each do |listener|
-        ActiveSupport::Notifications.unsubscribe listener if listener.instance_variable_get('@delegate') == subscriber
+        ActiveSupport::Notifications.unsubscribe listener if built_in_rails_listener?(listener, component)
       end
     end
   end
 
+  def built_in_rails_listener?(listener, component)
+    delegate = listener.instance_variable_get('@delegate')
+
+    # Assume that anything nested under the component namespace is built in to Rails
+    delegate.class.name.start_with?("#{component.to_s.classify}::")
```

Comment:
I don't think this is great, but I can't see a better way to check if the delegate is `subscriber` or (for Rails 7.1+) an instance of `ActionView::LogSubscriber::Start` for `ActionView` or `subscriber` for `ActionController`